templates/core/terraform/appgateway/certificate.tf

```diff
@@ -16,25 +16,43 @@ resource "azurerm_key_vault_certificate" "tlscert" {
-    password = "0000000000"
-  }
-
   certificate_policy {
     issuer_parameters {
       name = "Self"
     }
+
     key_properties {
       key_size   = 2048
       exportable = true
       key_type   = "RSA"
       reuse_key  = false
     }
+
     secret_properties {
       content_type = "application/x-pkcs12"
     }
+
+    x509_certificate_properties {
+      # Server Authentication = 1.3.6.1.5.5.7.3.1
+      # Client Authentication = 1.3.6.1.5.5.7.3.2
+      extended_key_usage = ["1.3.6.1.5.5.7.3.1"]
+
+      key_usage = [
+        "cRLSign",
+        "dataEncipherment",
+        "digitalSignature",
+        "keyAgreement",
+        "keyCertSign",
+        "keyEncipherment",
+      ]
+
+      subject_alternative_names {
+        dns_names = ["internal.contoso.com", "domain.hello.world"]
```

Comment:
do we need subject alternate names?

Reply:
Apparently we do, if we want to trust this cert on our local machines. Otherwise I get this: 

![image](https://user-images.githubusercontent.com/52954123/172899288-4df29750-2e25-4ae0-b98c-9fdd27a63230.png)